openedx/features/custom_fields/multiselect_with_other/db/fields.py

```diff
@@ -93,8 +93,9 @@ def to_python(self, value):
             if isinstance(value, list):
                 return value
             elif isinstance(value, string_type):
-                choices_str = value.replace(self.other_delimiter, '')
-                selected_choices = [choice for choice in choices_str.split(',') if choice.strip()]
+                choices_str = value.split(self.other_delimiter)
+                selected_choices = [choice for choice in choices_str[0].split(',') if choice.strip()]
+                selected_choices.append(choices_str[1])
```

Comment:
can we add a length check here just in case?

Reply:
We aren't concerned with content of string here, as we are splitting so it won't be None and if its lenth is zero it will append empty string which won't effect our output. If I'm missing any point please let me know.

Reply:
if the string is empty, and you split it on the base of anything, the resulting list will only have one index. and so the choices_str[1] will create an issue. whereas you're right choices_str[0] wont create an issue

Reply:
hence i was asking if we can add a check just before we access the "1" index

Reply:
I think Saad is right. If there is an empty string after the other_delimeter, choices_str[1] will give an index out of range error.

Reply:
Actually its our custom field where we always have pipe delimiter '|' so we would have 2 strings but since it become a concern here so I guess no harm in putting an extra check here.